include/circt/Dialect/HW/HWOpInterfaces.h

```diff
@@ -45,25 +45,70 @@ struct PortInfo : public ModulePort {
 /// This holds a decoded list of input/inout and output ports for a module or
 /// instance.
 struct ModulePortInfo {
-  explicit ModulePortInfo(ArrayRef<PortInfo> inputs, ArrayRef<PortInfo> outputs)
-      : inputs(inputs.begin(), inputs.end()),
-        outputs(outputs.begin(), outputs.end()) {}
-
-  explicit ModulePortInfo(ArrayRef<PortInfo> mergedPorts) {
-    inputs.reserve(mergedPorts.size());
-    outputs.reserve(mergedPorts.size());
-    for (auto port : mergedPorts) {
-      if (port.isOutput())
-        outputs.push_back(port);
-      else
-        inputs.push_back(port);
-    }
+  explicit ModulePortInfo(ArrayRef<PortInfo> inputs,
```

Comment:
I thought part of the reason to switch to ModuleType (name?) which is presumably the shift which this preparing for is to preserve module port ordering amongst inputs/outputs? If so, isn't sorting them here antithetical to that?

Reply:
This is just a storage format change for the current class, so it is preserving existing behavior.  Getting everything going through some iterators is the main point.

Reply:
If it's to facilitate efficient computation of `at_input`, based on a quick skim of these changes, I don't see any uses of `at_input` or `at_output`.

Reply:
shit... didn't see your comment. fucking gh.

Reply:
> This is just a storage format change for the current class, so it is preserving existing behavior. Getting everything going through some iterators is the main point.

I'm not noticing any places where inputs or outputs are randomly accessed... all the places you modified work through iterators, afaict.

Reply:
though preserving existing behavior for the sake of incrementalism is reasonable.

Reply:
|I'm not noticing any places where inputs or outputs are randomly accessed

There are a couple places where things are deleted.  Annoying,.